numpy/_core/tests/test_umath.py

```diff
@@ -2357,6 +2356,36 @@ def test_ldexp_overflow(self):
             assert_equal(ncu.ldexp(2., imax), np.inf)
             assert_equal(ncu.ldexp(2., imin), 0)
 
+    def test_ldexp_integer_scalar(self):
+        with warnings.catch_warnings():
+            # TODO: At the time of writing, this still raises a warning
+            # on certain platforms (see gh-29597). Once that is fixed,
+            # there is no need to catch warnings here.
+            warnings.filterwarnings('ignore', message='overflow encountered in ldexp',
+                                    category=RuntimeWarning)
+            warnings.filterwarnings('ignore', message='overflow encountered in cast',
+                                    category=RuntimeWarning)
```

Comment:
There may be a test not liking this.  We can use `pytest.mark.filterwarnings` here, I think.

(But let's see what is going on first, I guess...

Reply:
All tests pass now, but I can still push this if it's better!